src/main/scala/mirroring/services/databases/JdbcCTService.scala

```diff
@@ -118,17 +119,18 @@ object JdbcCTService extends LogSupport {
   }
 
   def getChangeTrackingVersion(query: String, jdbcContext: JdbcContext): BigInt = {
-    val jdbcService: JdbcService = new JdbcService(jdbcContext)
-    val jdbcDF: DataFrame        = jdbcService.loadData(query)
-
     var version: BigInt = BigInt(0)
-
-    if (!jdbcDF.isEmpty) {
-      version = BigInt(
-        jdbcDF
-          .collect()(0)
-          .getLong(0)
-      )
+    try {
+      version = spark.read
+        .format("jdbc")
+        .option("url", jdbcContext.url)
+        .option("dbtable", query)
+        .load()
+        .as[Long]
+        .first()
+    } catch {
```

Comment:
For the better experience,
```scala
case e: java.lang.NullPointerException =>
  logger.error(s"Change tracking is not enabled on queried table: ${query}")
  throw e
```
might be added.

Reply:
in what case there will be an NPE?

Reply:
There is NULL returned when CHANGE TRACKING is disabled on a DB or on a table.

Reply:
done